### PR TITLE
Plugin error handling

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py [hunk 1]
```diff
@@ -62,4 +77,8 @@ def _get_pot_via_script(self, script_path, visitor_data, data_sync_id):
         # the JSON response is always the last line
         script_data_resp = result.stdout.splitlines()[-1]
         self._logger.debug(f"_get_pot_via_script response = {script_data_resp}")
-        return json.loads(script_data_resp)['poToken']
+        try:
+            return json.loads(script_data_resp)['poToken']
+        except:
+            raise RequestError("Error parsing JSON response from _get_pot_via_script")
```
Comment (grqz) [inline plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:83]: is this necessary? `json.loads` should deal with all kinds of strings generated from `JSON.stringify`

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py [hunk 2]
```diff
@@ -27,22 +29,35 @@ def _get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data_sync_id=
...
+                'visitor_data': visitor_data,
+                'data_sync_id': data_sync_id
+            }).encode(), headers = {'Content-Type': 'application/json'}))
+        except Exception as e:
+            raise RequestError(f"Error reaching POST /get_pot: {str(e)}")
+            
+        try:
+            response_json = json.loads(response.read().decode('utf-8'))
+        except Exception as e:
+            raise RequestError(f"Error parsing response JSON: {str(e)}. response = {response.read().decode('utf-8')}")
```
Comment (grqz) [inline plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:46]: why not always send a json and put the error message into a field like `"error"`/`"errors"`. the failure to parse the response JSON isn't expected.

File: README.md [hunk 3]
```diff
@@ -99,5 +99,5 @@ If using option (a) HTTP Server for the provider, use yt-dlp like normal 🙂.
 If using option (b) script for the provider, you need to pass extractor arguments including the path to the generation script for each yt-dlp call. Make sure to point to the transpiled version, `server/build/generate_once.js`
```
Comment (grqz) [inline README.md:99]:
```suggestion
If using option (b) script for the provider, you need to pass the extractor argument `getpot_bgutil_script` to `youtube` for each yt-dlp call. The argument should include the path to the transpiled generation script (`server/build/generate_once.js`). E.g. `--extractor-args "youtube:getpot_bgutil_script=$WORKSPACE/bgutil-ytdlp-pot-provider/server/build/generate_once.js"`
```
(please also remove the codeblock below if taking this suggestion)
maybe?

**Review by grqz — CHANGES_REQUESTED** [review]

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py (hunk 1 above)
Comment (Brainicism) [reply plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:83]: Incase the output from the script is unexpected not a JSON.stringified value. Right now we expect that the JSON is the last line of stdout, if we accidentally print something to stdout after the JSON, we might have issues.

**Comment by grqz** [comment]
TODO: document: add extractor arg `getpot_bgutil_baseurl` and cmdline arg `-p` (or `--port`)

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py [hunk 4]
```diff
@@ -20,65 +21,71 @@ def _validate_get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data
 
     def _get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data_sync_id=None, player_url=None, **kwargs) -> str:
         generate_pot_script_path = ydl.get_info_extractor('Youtube')._configuration_arg('getpot_bgutil_script', [None], casesense=True)[0]
+        http_base_url = ydl.get_info_extractor('Youtube')._configuration_arg('getpot_bgutil_baseurl', ['http://127.0.0.1:4416'], casesense=True)[0]
```
Comment (Brainicism) [inline plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:24]: Are there any cases where the server wouldn't be running on the loopback address? 

My understanding is that the PO token has to be generated on the same IP address as the original request, so it should always be on 127.0.0.1? In that case, the only variable thing is probably the port number?

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py [hunk 5]
```diff
@@ -20,65 +21,71 @@ def _validate_get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data
...
-            po_token = self._get_pot_via_http(ydl, client, visitor_data, data_sync_id)
+            self._logger.info('Generating POT via HTTP server')
+            po_token = self._get_pot_via_http(ydl, client, visitor_data, data_sync_id, http_base_url)
 
         return po_token
 
-    def _get_pot_via_http(self, ydl, client, visitor_data, data_sync_id):
+    def _get_pot_via_http(self, ydl, client, visitor_data, data_sync_id, base_url):
         try:
             response = ydl.urlopen(Request('http://127.0.0.1:4416/get_pot', data=json.dumps({
```
Comment (Brainicism) [inline plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:36]: We're not using `base_url` here?

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py (hunk 4 above)
Comment (coletdjnz) [reply plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:24]: Could be running on a different IP on local network, or use docker container hostname/IP (likely be a common scenario if yt-dlp is also running in a docker container).

File: plugin/yt_dlp_plugins/extractor/getpot_bgutil.py (hunk 5 above)
Comment (grqz) [reply plugin/yt_dlp_plugins/extractor/getpot_bgutil.py:36]:
```suggestion
            response = ydl.urlopen(Request(f'{base_url}/get_pot', data=json.dumps({
```

**Review by grqz — APPROVED** [review]

**Pull request merged**